common-lib/src/cmr/common/concepts.clj

```diff
@@ -18,9 +18,15 @@
   (cset/map-invert generic-concept-types->concept-prefix))
 
 (defn get-generic-concept-types-array
-  "Gets the array of generic concept types."
-  []
-  (vec (keys generic-concept-types->concept-prefix)))
+  "Gets the array of generic concept types and optionally modify those values.
+   By default, no parameters will return the list as is, however if a function
+   is passed in that takes one value, this value will be changed in some way.
+   Expected usage is to pass in pluralize-concept-type."
+  ([] (get-generic-concept-types-array identity))
```

Comment:
I don't understand this line.  What is identity set to?  Is it defined somewhere else or does nil get passed in?

Reply:
**identity** is a single-arg function that simply returns its argument so is that shorthand for something like the default function https://clojuredocs.org/clojure.core/identity

Reply:
It seems like extra cpu cycles in most cases and by looking at the docs, it doesn't look like identity is being used in a standard way.  While identity is a function, what is the function with no arguments? is it a no-op?

I still think it would be more readable if the function were 
([] (keys generic-concept-types->concept-prefix))
([modifier-func]
 (reduce (fn [coll, item] (conj coll (modifier-func item)))
           []
           (keys generic-concept-types->concept-prefix))))

Reply:
I can cut out the keys:

> (reduce (fn [coll item] (conj coll (modifier-func (first item))))
          []
          generic-concept-types->concept-prefix))